libmozevent/mercurial.py

```diff
@@ -305,6 +307,24 @@ async def run(self):
                     "Unsupported repository", repo=build.repo_phid, build=build
                 )
 
+    def is_commit_skippable(self, build):
+        def get_files_touched_in_diff(rawdiff):
+            patched = []
+            for parsed_diff in rs_parsepatch.get_diffs(rawdiff):
+                # filename is sometimes of format 'test.txt  Tue Feb 05 17:23:40 2019 +0100'
+                if "filename" in parsed_diff:
+                    filename = parsed_diff["filename"].split(" ")[0]
+                    patched.append(filename)
+            return patched
+
+        for rev in build.stack:
+            files = get_files_touched_in_diff(rev.patch)
+            for patched_file in files:
+                if patched_file in self.skippable_files:
+                    # early return because one file touched is enough for us to bail out
+                    return True
+        return False
```

Comment:
Nicer Pythonic way:
```
return any(patched_file in self.skippable_files for rev in build.stack for patched_file in get_files_touched_in_diff(rev.patch))
```

Reply:
Not sure I would agree with the double-loop as being more readable, but happy to take that. Updated PR coming.